### DISCUSSION TITLE
Happy upgrade — thanks for such an incredibly stable system

### DISCUSSION THREAD

**Discussion by Eric38fr** [Homebrew/discussions#5218] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.14
[...]
```

### Output of `brew doctor`

```shell
Warning: You are using macOS 10.15.
We (and Apple) do not provide support for this old version.
It is expected behaviour that some formulae will fail to build in this old version.
It is expected behaviour that Homebrew will be buggy and slow.
[...]
```

### Description of issue

Just a short report for a happy upgrade. I am still working with macos 10.15.9. Wanting to install "ffmpeg" with brew, I realize that I have never updated my brew since I installed it in april 2020, version 2.2.13 ! However, I try: `brew update` ➥ it stops at 56% during the first download, with "_curl: (56) LibreSSL SSL_read: SSL_ERROR_SYSCALL, errno 54_", and a few error messages later, and some advices (though it never said "_you should upgadre your os_", I can feel the intent…), the final diagnostics : "_Failed to install Homebrew Portable Ruby (and your system version is too old)!_" ! :-(((

So now, I try to anticipate future problems with this old-out install and look for which added stuff are really essential to me : cd `brew --cache`; ls ; deletion of a specific file as adviced…  Then my fingers type without me thinking at it `brew --version` : 4.2.14 !!! UPDATED ! and working !!! A `brew doctor`gives me a few more advices, a `brew cleanup`, a `brew untap …`, and a last `brew update`= "Already up-to-date" : WAOUH !!!

Many thanks to all the brew team, for having designed such a wonderful product, with this incredible stability, that you can forget upgrades for FOUR YEARS, and still it works perfectly well (while on my smartphone, some apps work with a quite daily upgrade, and if you miss them for a month, do not works at all !). Best regards -- Éric38fr

**Comment by razvanazamfirei** [comment]
Thank you for the kind words, @Eric38fr! ❤️
